1_wqp_pull/src/wqp_inventory.R

```diff
@@ -71,86 +71,102 @@ inventory_wqp <- function(state_id, wqp_pull_params) {
+
+
   # catch errors, and break up data into two calls as a backup plan
   # this splits the call into half by counties
-  if(!'list' %in% class(temp_dat)) {
-    message('State call failed, calling by 1/2 of counties at a time.')
-    state_counties <- filter(dataRetrieval::countyCd, STATE %in% gsub('US:', '', state_id))
-    
-    # split counties in half and make a 2-part call
-    split <- ceiling(nrow(state_counties)/2)
-    
-    # first half of counties
-    wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
-    temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
-    
-    # second half of counties
-    wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
-    temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
-    
-    # combine outputs
-    temp_dat <- list(time = temp1$time + temp2$time,
-                     nrow = nrow(temp1$out) + nrow(temp2$out),
-                     out = bind_rows(temp1$out, temp2$out))
-  }
-  
+  # if(!'list' %in% class(temp_dat)) {
+  #   message('State call failed, calling by 1/2 of counties at a time.')
+  #   state_counties <- filter(dataRetrieval::countyCd, STATE %in% gsub('US:', '', state_id))
+  #
+  #   # split counties in half and make a 2-part call
+  #   split <- ceiling(nrow(state_counties)/2)
+  #
+  #   # first half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[1:split]
+  #   temp1 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # second half of counties
+  #   wqp_args$countycode <- paste('US', state_counties$STATE, state_counties$COUNTY, sep = ':')[(split+1):nrow(state_counties)]
+  #   temp2 <- wqp_call(whatWQPdata, wqp_args[c('characteristicName', 'countycode')])
+  #
+  #   # combine outputs
+  #   temp_dat <- list(time = temp1$time + temp2$time,
+  #                    nrow = nrow(temp1$out) + nrow(temp2$out),
+  #                    out = bind_rows(temp1$out, temp2$out))
+  # }
+
   # summarize what was retrieved and how long it took
   message('Retrieved ', temp_dat$nrow, ' rows of data in ', ceiling(temp_dat$time), ' seconds.')
-  
+
   return(temp_dat$out)
 }
 combine_inventory <- function(ind_file, ...) {
   dat_out <- bind_rows(...)
-  
+
   # keep columns of interest, put back lat/long
-  # keeping lat/long here in case there was a situation where you'd want to 
+  # keeping lat/long here in case there was a situation where you'd want to
   # take stock/map sites before pulling data
   dat_out <- dat_out %>%
-    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName, 
+    select(OrganizationIdentifier, MonitoringLocationIdentifier, ResolvedMonitoringLocationTypeName,
            StateName, CountyName, HUCEightDigitCode, latitude = lat, longitude = lon, resultCount)
```

Comment:
The data pull is based on how many records come from each site. Now, we have the same site in multiple rows since sites we're queried through time. Here, you'll have to reduce the combined data down so that the resultCount reflects the sum of observations for each site.